addons/binding/org.openhab.binding.gpstracker/src/main/java/org/openhab/binding/gpstracker/internal/GPSTrackerHandlerFactory.java

```diff
@@ -134,6 +142,13 @@ protected void removeHandler(ThingHandler thingHandler) {
     protected void activate(ComponentContext componentContext) {
         super.activate(componentContext);
 
+        //get system location
+        ServiceReference<LocationProvider> locServiceReference = bundleContext.getServiceReference(LocationProvider.class);
```

Comment:
Why do you use the `bundleContext` here and not simply have the `LocationProvider` injected through a setter method like all other services?